File: easy/product_of_array_except_self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        output = [1] * len(nums)
        productFromLeft = 1
        for index in range(len(nums)):
            output[index] *= productFromLeft
            productFromLeft *= nums[index]
        productFromRight = 1
        for index in range(len(nums)-1, -1, -1):
            output[index] *= productFromRight
            productFromRight *= nums[index]
        return output
        

File: medium/product_of_array_except_self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> output(nums.size(), 1);
        int productFromLeft = 1;
        for(int i = 0; i < nums.size(); i++) {
            output[i] *= productFromLeft;
            productFromLeft *= nums[i];
        }
        int productFromRight = 1;
        for(int j = nums.size()-1; j >= 0; j--) {
            output[j] *= productFromRight;
            productFromRight *= nums[j];
        }
        return output;
    }
};
